### PR TITLE
UofT-DSI | Deep Learning - Assignment 2

### PR DESCRIPTION
**Pull request by asnjunior**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Submitting assignment 2

## What did you learn from the changes you have made?
How to build a text generating model

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
NA

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
My accuracy score is low and can only improve it by a few percentage points. This is already using pre-trained embeddings. The best I can do is 7%.

## How were these changes tested?
NA

## A reference to a related issue in your repository (if applicable)
NA

## Checklist
- [ x ] I can confirm that my changes are working as intended

### CODE REVIEW

File: 01_materials/labs/lab_2.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_2.ipynb:1]: Can you explain in a few words why the best accuracy you got in the last example is lower than the one you achieved before?

File: 01_materials/labs/lab_3.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_3.ipynb:1]: Can you explain what you have changed in the final exercise? It seems to be the same as the previous baseline model.

File: 01_materials/labs/lab_4.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_4.ipynb:1]: The linear scaling of the resized image seems unnecessary to me `resize(image, (224, 224), mode='reflect', anti_aliasing=True) *224`

File: 02_activities/assignments/assignment_2.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_2.ipynb:1]: In the attached code cell, you defined the `OOV_TOKEN` as an integer and pass it to the `Tokenizer` via the `num_words` arguments. `OOV_TOKEN` is used to represent unknown or out-of-vocab words. Could you please adjust the code accordingly and rerun the remaining part of the notebook?
```
from keras.preprocessing.text import Tokenizer

# Your code here
VOCAB_SIZE = 2960
OOV_TOKEN = 50

# vectorize the text samples into a 2D integer tensor
tokenizer = Tokenizer(num_words=OOV_TOKEN, char_level=False)
tokenizer.fit_on_texts([text_lc])

word_index = tokenizer.word_index
print('Found %s unique tokens.' % len(word_index))
print(f'Example of word_index: {list(word_index.items())[:]}')
```
Also, please generate a few words with the refined model.

**Review by tianyi21 — CHANGES_REQUESTED** [review]
Hi Alfredo!

Thank you for submitting your assignments. 

Please revise and resubmit your Lab3 and A2 following my comments. If you have any questions, please don't hesitate to let us know.

Thanks,
Tianyi [LS]

**Comment by asnjunior** [comment]
Hi Tianyi. Noted. I will revise and resubmit later tonight.

**Comment by asnjunior** [comment]
Hi Tianyi. I am done updating my Lab 3 and Assignment 2. You also had a comment for Lab 2 above, should I also work on that?

File: 02_activities/assignments/assignment_2.ipynb
Comment (tianyi21) [inline 02_activities/assignments/assignment_2.ipynb:1]: Thank you for revising your submission. Your A2 is now complete. Just for your reference, I think it is not common to use `Conv1D` layer here. If you want to further improve your model performance, you could try adding additional `LSTM` layers.

File: 01_materials/labs/lab_3.ipynb
Comment (tianyi21) [inline 01_materials/labs/lab_3.ipynb:1]: Thank you for revising your Lab workbook. Your lab workbooks are now complete. However, it seems to me that you declared additional `dense` layers without `Dropout` but didn't utilize them in the `call` function.

**Review by tianyi21 — APPROVED** [review]
Hi Alfredo!

Thank you for revising your DL assignments.

All your DL assignments have been completed. Thank you for your participation, and I wish you all the best for your future!

Thanks,
Tianyi [LS]
